### PR TITLE
SQL: introduce SQLite storage

### PR DESCRIPTION
**Pull request by reinkrul**
Part of https://github.com/nuts-foundation/nuts-node/issues/2340

### CODE REVIEW

File: storage/engine.go [hunk 1]
```diff
@@ -129,6 +152,49 @@ func (e *engine) GetSessionDatabase() SessionDatabase {
...
+	e.sqlDB, err = gorm.Open(sqlite.Open(connectionString), &gorm.Config{})
+	if err != nil {
+		return err
+	}
+	log.Logger().Debug("Running database migrations...")
+	underlyingDB, err := e.sqlDB.DB()
+	if err != nil {
+		return err
+	}
+	sourceDriver, err := iofs.New(sqlMigrationsFS, "sql_migrations")
```
Comment (woutslakhorst) [inline storage/engine.go:177]: How will the correct migrations be selected in the future when there's only a connection string? eg: mysql vs postgresql

File: storage/interface.go [hunk 2]
```diff
@@ -37,6 +38,8 @@ type Engine interface {
 	GetProvider(moduleName string) Provider
 	// GetSessionDatabase returns the SessionDatabase
 	GetSessionDatabase() SessionDatabase
+
+	SQLDatabase() *gorm.DB
```
Comment (woutslakhorst) [inline storage/interface.go:42]: godoc, other methods are prepended with `Get`. All `Get` or no `Get`?

File: storage/sql_migrations/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+This directory contains SQL schema migrations, run at startup of the node.
+
+Refer to https://github.com/golang-migrate/migrate/blob/master/MIGRATIONS.md on how to write migrations.
+
+Files should be named according to the following: `<number>_<short engine name>_<table/feature name>.<up|down>.sql`.
+For instance: `2_usecase_list.up.sql`.
+
+Remember NOT to alter migrations that were included in released versions of the Nuts node: it might break vendor deployments or cause data corruption.
```
Comment (woutslakhorst) [inline storage/sql_migrations/README.md:8]: important for instances running master. It'll break.

File: storage/engine.go (hunk 1 above)
Comment (reinkrul) [reply storage/engine.go:177]: probably connection string driver name to folder

File: storage/sql_migrations/README.md (hunk 3 above)
Comment (reinkrul) [reply storage/sql_migrations/README.md:8]: Maybe alter to "avoid changing migrations in master, DO NOT alter migrations in a released version". E.g., to fix a broken migration (not just one that is revised)?

File: storage/interface.go (hunk 2 above)
Comment (reinkrul) [reply storage/interface.go:42]: In Go you generally leave out "Get" for "GetXXX", but we should probably do that in one "go". Will change to "GetSQLDatabase()" for the sake of consistency

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**